packages/ui/src/BlogAppbar.tsx

```diff
@@ -30,7 +30,7 @@ export const BlogAppbar = ({
   }, [track, problem]);
 
   let totalPages = Array.from({ length: track.problems.length }, (_, i) => i + 1);
-
+  localStorage.setItem('problemIndex', (problemIndex+1).toString());
```

Comment:
Why here?
Also why localStorage?

Reply:
Figured we could use problemIndex in other places so why not have it on hand in the parent component i.e BlogAppbar.